### PR TITLE
Video-18-Create-Sample-Users-In-MongoDB

### PR DESCRIPTION
**Pull request by basir**

### CODE REVIEW

File: backend/server.js [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
+import mongoose from 'mongoose';
 import data from './data.js';
+import userRouter from './routers/userRouter.js';
 
 const app = express();
+mongoose.connect(process.env.MONGODB_URL || 'mongodb://localhost/amazona', {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useCreateIndex: true,
+});
```
Comment (ashiqdammam) [inline backend/server.js:11]: useNewUrlParser, useUnifiedTopology, useFindAndModify, and useCreateIndex are no longer supported options....Removed

Comment (longnguyen2210) [reply backend/server.js:11]: You only need to remove "useCreateIndex: true", and the code would work perfectly fine.

File: backend/server.js [hunk 2]
```diff
@@ -1,7 +1,14 @@
 import express from 'express';
+import mongoose from 'mongoose';
 import data from './data.js';
+import userRouter from './routers/userRouter.js';
 
 const app = express();
+mongoose.connect(process.env.MONGODB_URL || 'mongodb://localhost/amazona', {
+  useNewUrlParser: true,
```
Comment (quavo19) [inline backend/server.js:8]: 1)go to the control panel

2)Select System and Security and choose the last option: Administrative tools.

3)Select Services

4)Search for MongoDB Server (MongoDB)

5) Right-click on it and click on the run.

6)and close the Services blindly

7)last but not least, change your MongoDB URL to mongodb://0.0.0.0:27017/

8)Restart your server and that worked for me!

**Review by sridharsiddu224 — COMMENTED** [review]
**showing error like below**

node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::5000
    at Server.setupListenHandle [as _listen2] (node:net:1485:16)
    at listenInCluster (node:net:1533:12)
    at Server.listen (node:net:1621:7)
    at Function.listen (E:\Reactjs It\Mkart\mkart\node_modules\express\lib\application.js:635:24) 
    at file:///E:/Reactjs%20It/Mkart/mkart/backend/server.js:42:5
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1512:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'EADDRINUSE',
  errno: -4091,
  syscall: 'listen',
  address: '::',
  port: 5000
}
Node.js v18.12.1
[nodemon] starting `node --experimental-modules backend/server.js`
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::5000
    at Server.setupListenHandle [as _listen2] (node:net:1485:16)
    at listenInCluster (node:net:1533:12)
    at Server.listen (node:net:1621:7)
    at Function.listen (E:\Reactjs It\Mkart\mkart\node_modules\express\lib\application.js:635:24) 
    at file:///E:/Reactjs%20It/Mkart/mkart/backend/server.js:43:5
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1512:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'EADDRINUSE',
  errno: -4091,
  syscall: 'listen',
  address: '::',
  port: 5000
}

Node.js v18.12.1
[nodemon] app crashed - waiting for file changes before starting...

**Comment by sridharsiddu224** [comment]
if possible please give me sugestion

**Comment by quavo19** [comment]
> if possible please give me suggestion

@sridharsiddu224 the error says the address in use
so you have to stop all activities on port 5000 and start the server again

contact me @[Donald](https://www.linkedin.com/in/donaldakite/)

**Comment by sridharsiddu224** [comment]
how to stop server please suggest me 

> > if possible please give me suggestion
> 
> @sridharsiddu224 the error says the address in use so you have to stop all activities on port 5000 and start the server again
> 
> contact me @[Donald](https://www.linkedin.com/in/donaldakite/)

**Comment by sridharsiddu224** [comment]
Ok thankyou

**Pull request merged**